spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/discovery/DefaultKubernetesServiceInstance.java

```diff
@@ -101,6 +106,16 @@ public String getCluster() {
 		return this.cluster;
 	}
 
+	@Override
+	public Map<String, String> podLabels() {
```

Comment:
Is there a reason to not just put this in the metadata?

Reply:
I was afraid that service labels and pod labels could create a mess if put together, like what if labels colide? i thought that separating them would be cleaner.

Reply:
Yeah that is kind of what I was saying here....I suppose there is no other way, we either run the risk of colliding or we separate things.

Since we have some freedom here with this new property could we make it Map<String, Map<String, String> call it `podMetadata` and put both labels and annotations in that map?

Reply:
my turn to miss-read!

```
Map<String, Map<String, String>
```

so something like : `labels = {a = b}` and `annotations = {c = d}`, right?

Reply:
correct